tests/scripts/analyze_outcomes.py

```diff
@@ -693,22 +685,17 @@ def main():
 
         KNOWN_TASKS['analyze_coverage']['args']['full_coverage'] = options.full_coverage
 
-        all_succeeded = True
-
         for task in tasks_list:
             test_function = KNOWN_TASKS[task]['test_function']
             test_args = KNOWN_TASKS[task]['args']
-            test_log = test_function(options.outcomes, test_args)
-            # Merge the output of this task with the main one
-            main_log.output = main_log.output + test_log.output
-            main_log.info("Task {} completed with:\n".format(task) + \
-                            "{} warnings\n".format(test_log.warning_count) + \
-                            "{} errors\n".format(test_log.error_count))
-            if test_log.error_count != 0:
-                all_succeeded = False
+            main_log = test_function(main_log, options.outcomes, test_args)
+        
+        main_log.info("Overall results:\n" + \
+                      "{} warnings\n".format(main_log.warning_count) + \
+                      "{} errors\n".format(main_log.error_count))
 
         main_log.print_output()
-        sys.exit(0 if all_succeeded else 1)
+        sys.exit(0 if (main_log.error_count == 0) else 2)
```

Comment:
I think this should be `else 1` as per Gilles's suggestion: 0 means success, 1 means one or more tests failed, >= 2 means something else went wrong (like, wrong command line arguments).

Reply:
The "fun fact" is that I even looked at that comment before changing this, but apparently I didn't really paid attentions at what I read :/